src/pybind/mgr/dashboard/frontend/src/app/shared/services/page-visibility-helper.ts

```diff
@@ -0,0 +1,20 @@
+import { fromEvent, Observable, partition } from 'rxjs';
+import { repeatWhen, shareReplay, takeUntil } from 'rxjs/operators';
+
+export function whenPageVisible() {
```

Comment:
Not sure if this (src/app/shared/services/page-visibility-helper.ts) is the best file location for this custom RxJS operator

Reply:
Just for consistency with the rxjs directory, what about below `.../shared/rxjs/operators`?

Reply:
Sounds good  to me :+1: